Logger/BaseLogger.cs

```diff
@@ -1,7 +1,23 @@
-﻿namespace Logger
+﻿using System;
+
+namespace Logger
 {
     public abstract class BaseLogger
     {
+        public string ClassName { get; set; }
         public abstract void Log(LogLevel logLevel, string message);
+
+        protected string FormatMsg(LogLevel logLevel, string message)
+        {
+            /*
+ The current date/time
+The name of the class that created the logger
+The log level
+The message
+The format may vary, but an example might look like this "10/7/2019 12:38:59 AM FileLoggerTests Warning: Test message" 
+*/
+            return $"{DateTime.Now} {ClassName} {logLevel} {message}\n";
```

Comment:
When dealing with code that runs on both Windows and Mac we have to deal with the subtle differences in line endings (`\r\n` on Windows and `\n` on everything else). To account for this difference you can use `Environment.NewLine` which will give you the correct one depending on which operating system your code is running on.
```suggestion
            return $"{DateTime.Now} {ClassName} {logLevel} {message}{Environment.NewLine}";
```